packages/devextreme/js/ui/html_editor.d.ts

```diff
@@ -46,7 +46,150 @@ export type HtmlEditorImageUploadTab = 'url' | 'file';
+export type HtmlEditorPredefinedToolbarItem = 'background' | 'bold' | 'color' | 'font' | 'italic' | 'link' | 'image' | 'size' | 'strike' | 'subscript' | 'superscript' | 'underline' | 'blockquote' | 'header' | 'increaseIndent' | 'decreaseIndent' | 'orderedList' | 'bulletList' | 'alignLeft' | 'alignCenter' | 'alignRight' | 'alignJustify' | 'codeBlock' | 'variable' | 'separator' | 'undo' | 'redo' | 'clear' | 'cellProperties' | 'tableProperties' | 'insertTable' | 'insertHeaderRow' | 'insertRowAbove' | 'insertRowBelow' | 'insertColumnLeft' | 'insertColumnRight' | 'deleteColumn' | 'deleteRow' | 'deleteTable' | 'ai';
+
+/** @public */
+export type HtmlEditorAICommandName = 'summarize' | 'proofread' | 'expand' | 'shorten' | 'changeStyle' | 'changeTone' | 'translate' | 'askAI';
+
+/** @public */
+export type HtmlEditorAIChangeStyleOption =
+    | 'formal'
+    | 'informal'
+    | 'technical'
+    | 'business'
+    | 'creative'
+    | 'journalistic'
+    | 'academic'
+    | 'persuasive'
+    | 'narrative'
+    | 'expository'
+    | 'descriptive'
+    | 'conversational';
+
+/** @public */
+export type HtmlEditorAIChangeToneOption =
+    | 'professional'
+    | 'casual'
+    | 'straightforward'
+    | 'confident'
+    | 'friendly';
+
+/** @public */
+export type HtmlEditorAITranslateOption =
+    | 'arabic'
+    | 'chinese'
+    | 'english'
+    | 'french'
+    | 'german'
+    | 'japanese'
+    | 'spanish';
+
+export type HtmlEditorAICommandNameAll = HtmlEditorAICommandName | 'custom';
+
+/**
+ * @docid
+ * @type object
+ * @namespace DevExpress.ui.dxHtmlEditor
+ */
+export interface HtmlEditorAICommandBase<CommandName extends HtmlEditorAICommandNameAll, CommandOptions = undefined> {
+    name: CommandName;
+    text?: string;
+    options?: CommandOptions;
+}
+
+/**
+ * @docid
+ * @public
+ * @type object
+ * @inherits HtmlEditorAICommandBase
+ * @namespace DevExpress.ui.dxHtmlEditor
+ */
+export type HtmlEditorAIBasicCommand =
+    | HtmlEditorAICommandBase<'summarize', never>
+    | HtmlEditorAICommandBase<'proofread', never>
+    | HtmlEditorAICommandBase<'expand', never>
+    | HtmlEditorAICommandBase<'shorten', never>
+    | HtmlEditorAICommandBase<'askAI', never>;
+
+/**
+ * @docid
+ * @public
+ * @type object
+ * @inherits HtmlEditorAICommandBase
+ * @namespace DevExpress.ui.dxHtmlEditor
+ */
+export type HtmlEditorAIChangeStyleCommand = HtmlEditorAICommandBase<'changeStyle', HtmlEditorAIChangeStyleOption[]>;
```

Comment:
Should be `| string` here?
I don't see where an ability for specifying custom param is supported

Reply:
I'm not sure it's optimal but will see what API reviewers say